refactor(test): improve L1ERC721Bridge test structure and documentation

aliersh opened the pull request:
- consolidate test initialization into a single L1ERC721Bridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- convert @dev to @notice tags where appropriate
- ensure comment lines stay within 100 character limit
- move test_pause_finalizeBridgeERC721_reverts into FinalizeBridgeERC721 contract
- rename test_pause_finalizeBridgeERC721_reverts to test_finalizeBridgeERC721_paused_reverts
- inline PauseFail contract's setup directly into the renamed test function

This PR replaces #15596 

Rebuilt the L1ERC721Bridge test refactor from scratch on top of the latest develop to avoid rebase complexity. All relevant structural and documentation improvements have been re-applied cleanly.

aliersh committed:
refactor(test): improve L1ERC721Bridge test structure and documentation

- consolidate test initialization into a single L1ERC721Bridge_TestInit contract
- organize individual test functions into separate contracts inheriting from TestInit
- add @title and @notice tags to all test contracts
- convert @dev to @notice tags where appropriate
- ensure comment lines stay within 100 character limit
- move test_pause_finalizeBridgeERC721_reverts into FinalizeBridgeERC721 contract
- rename test_pause_finalizeBridgeERC721_reverts to test_finalizeBridgeERC721_paused_reverts
- inline PauseFail contract's setup directly into the renamed test function

smartcontracts commented on packages/contracts-bedrock/test/L1/L1ERC721Bridge.t.sol:
```diff
@@ -127,8 +137,226 @@ contract L1ERC721Bridge_Test is CommonTest {
+    /// @dev Checks that the `paused` state of the bridge matches the `paused` state of the
+    ///      `superchainConfig` after it's been changed.
+    function test_pause_matchesSuperchainConfig_succeeds() external {
+        assertFalse(l1StandardBridge.paused());
+        assertEq(l1StandardBridge.paused(), systemConfig.paused());
+
+        vm.prank(superchainConfig.guardian());
+        superchainConfig.pause(address(0));
+
+        assertTrue(l1StandardBridge.paused());
+        assertEq(l1StandardBridge.paused(), systemConfig.paused());
```
Correct, should be changed, this was present in the original test, the refactor just made it evident. Good bot

aliersh replied:
Done — committed in [`ff1d865`](https://github.com/ethereum-optimism/optimism/pull/15735/commits/ff1d865ceaa56c4c3575d1517a6a4cc7f1da984b)

smartcontracts commented on packages/contracts-bedrock/test/L1/L1ERC721Bridge.t.sol:
```diff
@@ -127,8 +137,226 @@ contract L1ERC721Bridge_Test is CommonTest {
+
+    /// @notice Ensures that the `bridgeERC721` function reverts when the bridge is paused.
+    function test_finalizeBridgeERC721_paused_reverts() external {
+        /// @dev Sets up the test by pausing the bridge, giving ether to the bridge and mocking
```
Just use a normal comment, no need for `/// @dev` here

aliersh replied:
Done — committed in [ff1d865](https://github.com/ethereum-optimism/optimism/pull/15735/commits/ff1d865ceaa56c4c3575d1517a6a4cc7f1da984b)

aliersh committed:
fix(test): fix bridge reference and remove @dev tag in L1ERC721Bridge tests

- Replace l1StandardBridge with l1ERC721Bridge in test_pause_matchesSuperchainConfig_succeeds test assertions
- Remove @dev tag from inside test_finalizeBridgeERC721_paused_reverts function test comment

smartcontracts commented:
/ci authorize ff1d865ceaa56c4c3575d1517a6a4cc7f1da984b
